components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/connector/CoordinatedTaskDBConnector.java

```diff
@@ -0,0 +1,362 @@
+import org.apache.commons.logging.LogFactory;
+import org.wso2.micro.integrator.ntask.coordination.task.ConstantUtil;
+import org.wso2.micro.integrator.ntask.coordination.task.CoordinateTaskRunTimeException;
+import org.wso2.micro.integrator.ntask.coordination.task.CoordinatedTask;
+import org.wso2.micro.integrator.ntask.coordination.task.db.TaskQueryHelper;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+
+/**
+ * The connector class which deals with underlying coordinated task table.
+ */
+public class CoordinatedTaskDBConnector {
+
+    private static final Log LOG = LogFactory.getLog(CoordinatedTaskDBConnector.class);
+    private DataSource dataSource;
+
+    /**
+     * Constructor.
+     *
+     * @param dataSource - The datasource config to initiate the connection.
+     * @throws SQLException - Exception.
+     */
+    public CoordinatedTaskDBConnector(DataSource dataSource) throws SQLException {
+
+        this.dataSource = dataSource;
+        try (Connection connection = getConnection()) {
+            DatabaseMetaData metaData = connection.getMetaData();
+            String databaseType = metaData.getDatabaseProductName();
+            if (!"MySQL".equals(databaseType)) {
+                throw new CoordinateTaskRunTimeException(
+                        "Not supported data base type found : " + databaseType + " . Only MySql is supported.");
+            }
+        }
+    }
+
+    /**
+     * Removes the node id of the task and update the task state.
+     *
+     * @param coordinatedTasks - List of coordinated tasks which needs to be updated.
+     */
+    public void cleanInvalidNodesAndUpdateState(List<CoordinatedTask> coordinatedTasks) throws SQLException {
+
+        if (coordinatedTasks.isEmpty()) {
+            return;
+        }
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.CLEAN_INVALID_NODES)) {
+            for (CoordinatedTask task : coordinatedTasks) {
+                String taskName = task.getTaskName();
+                preparedStatement.setString(1, task.getTaskState().name());
+                preparedStatement.setString(2, taskName);
+                preparedStatement.addBatch();
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Removing the node assignment of task [" + taskName + "].");
+                }
+            }
+            preparedStatement.executeBatch();
+        }
+    }
+
+    /**
+     * For all the tasks which has this destined node id , sets it to null and update the task state to none if it was
+     * in a running state.
+     *
+     * @param nodeId - Node Id.
+     */
+    public void cleanTasksOfNode(String nodeId) throws SQLException {
```

Comment:
Again let's be explicit about what this method does. Maybe 'unassignRunningTasksAndUpdateStateToNone'